r/R/table.R

```diff
@@ -248,39 +203,9 @@ Table$create <- function(..., schema = NULL) {
   }
 }
 
-#' @export
-as.data.frame.Table <- function(x, row.names = NULL, optional = FALSE, ...) {
-  df <- Table__to_dataframe(x, use_threads = option_use_threads())
-  if (!is.null(r_metadata <- x$metadata$r)) {
-    df <- apply_arrow_r_metadata(df, .unserialize_arrow_r_metadata(r_metadata))
-  }
-  df
-}
-
-#' @export
-as.list.Table <- as.list.RecordBatch
-
-#' @export
-row.names.Table <- row.names.RecordBatch
-
-#' @export
-dimnames.Table <- dimnames.RecordBatch
-
-#' @export
-dim.Table <- function(x) c(x$num_rows, x$num_columns)
-
 #' @export
 names.Table <- function(x) x$ColumnNames()
 
-#' @export
-`names<-.Table` <- function(x, value) x$RenameColumns(value)
-
-#' @export
-`[.Table` <- `[.RecordBatch`
-
-#' @export
-`[[.Table` <- `[[.RecordBatch`
-
 #' @export
 `[[<-.Table` <- function(x, i, value) {
```

Comment:
Could you add a TODO/comment here describing why these have not yet been changed to be methods of the base class `ArrowTabular`? And another one where `$<-.Table` is defined

Reply:
I ended up moving them to ArrowTabular and left `stop(TODO)` messages in the missing RecordBatch methods.